### ISSUE TITLE
Add handle to field list on conditions and make field wider

### ISSUE THREAD

**Issue by roland-d** [verbb/formie#1005]
### What are you trying to do?

I am adding conditions to form fields but it is hard to find the correct field name because it only shows the label and not the handle. Further the dropdown is kind of small, so I do not always see the full name. This is what I see now:

![image](https://user-images.githubusercontent.com/359377/181026771-c3681d1b-b5fa-463d-8a3d-0584349d910d.png)

### What's your proposed solution?

I would like to see the handle behind the label. Something like `Name (child1_name)` so when you get a list you see this:
- Name (child1_name)
- Name (child2_name)
- Name (child3_name)

Perhaps an option to switch between the options so users have a choice. Default to Label to keep the current behavior.
- Label
- Handle
- Label (Handle)

**Comment by engram-design** [comment]
I'll consider this, although if you have a field with a long label, chances are the handle with also be quite long (unless you're in-the-know enough to change the handle to something more concise). For example I see the following field pretty commonly around:

- Enter your full name as shown on your license (enterYourFullNameAsShownOnYourLicense)

Which would be pretty long, especially if that was also a multi-name field:

- Enter your full name as shown on your license (enterYourFullNameAsShownOnYourLicense): Prefix
- Enter your full name as shown on your license (enterYourFullNameAsShownOnYourLicense): First Name

So there's things to consider. _Of course_ I would recommend people use a combination of label and instructions, but people have use-cases for long labels.

Would it help if you were able to search that dropdown like an autocomplete-style control?

**Comment by roland-d** [comment]
You are correct, if you have a long label, you will have a long handle. In my case I have set my own handles.

> Which would be pretty long, especially if that was also a multi-name field:

Yes but if we can set the option to show only the handle or only label for example, that would make it shorter. Hence my suggestion to be able to configure what a user wants to see, that way the user can decide for themselves what they want to see. Where the default is the label so it works the same as it does now.

> Of course I would recommend people use a combination of label and instructions, but people have use-cases for long labels.

Indeed, and the instruction field is also heavily used :) But it is not only long labels, there are labels like `Name` that is occurring a few times in my form, without the handle I have to guess which one I need.

> Would it help if you were able to search that dropdown like an autocomplete-style control?

Regardless of handle, that would be a nice feature. My form is 7 pages long as they have a lot of questions to ask. So searching the current dropdown is also challenging.

**Comment by engram-design** [comment]
Thanks for the feedback and ideas, I'll see what I can come up with!

**Comment by engram-design** [comment]
So just coming back to this.

> Further the dropdown is kind of small, so I do not always see the full name.

We limit the text shown in dropdowns to 42 characters, which just helps against long text in dropdowns causing layout issues. We also use this same dropdown control for mapping integration values, page conditions, button conditions and notification conditions. I have however moved this up to 60 characters.

> Perhaps an option to switch between the options so users have a choice.

I was going to put this setting in the general Formie > Settings > Forms, so you'd only be able to configure this site-wide. Will that suffice? I just note that this can't be toggled on-off at will.

> Would it help if you were able to search that dropdown like an autocomplete-style control?

Probably going to mark this as long-term, as most autocomplete controls I've found have pretty poor performance.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.1](https://github.com/verbb/formie/releases/tag/4.0.0-beta.1) now the handle will be shown, but only if a duplicate is detected, which helps lower the noise if there aren't any duplicates (there's no need to show the handle, really).
